react/src/components/molecules/LinkDropdown/index.js

```diff
@@ -0,0 +1,118 @@
+import DecorativeLink from '../../atoms/links/DecorativeLink';
+import Icon from '../../atoms/icons/Icon';
+import './style.css';
+
+class LinkDropdown extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      buttonExpand: false
+    };
+    this.wrapperRef = React.createRef();
+    this.setDropDownButtonRef = this.setDropDownButtonRef.bind(this);
+    this.closeDropdown = this.closeDropdown.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.handleSelect = this.handleSelect.bind(this);
+    this.handleClickOutside = this.handleClickOutside.bind(this);
+  }
+  componentDidMount() {
+    this.buttonClicked = false;
+    document.addEventListener('mousedown', this.handleClickOutside);
+    this.dropDownButtonRef.addEventListener('mousedown', this.handleRefMouseDown);
+  }
+  componentWillUnmount() {
+    document.removeEventListener('mousedown', this.handleClickOutside);
+    this.dropDownButtonRef.removeEventListener('mousedown', this.handleRefMouseDown);
+  }
+  setDropDownButtonRef(node) {
+    this.dropDownButtonRef = node;
+  }
+  closeDropdown() {
+    this.setState({ buttonExpand: false });
+  }
+  handleClick() {
+    this.setState((prevState) => ({ buttonExpand: !prevState.buttonExpand }));
+  }
+  handleSelect(event) {
+    const { target } = event;
+    this.setState({ buttonExpand: false });
+    if (is.fn(this.props.onItemSelect)) {
+      this.props.onItemSelect({ target });
+    }
+  }
+  handleClickOutside(event) {
+    // Close the panel if the user clicks outside the component.
+    const node = this.wrapperRef.current;
+    if ((node && !node.contains(event.target))) {
+      if (this.state.buttonExpand) {
+        this.setState({ buttonExpand: false });
+      }
+    }
+  }
+  render() {
+    const { dropdownItems, dropdownButton } = this.props;
+    const dropdownButtonProps = {
+      ...dropdownButton,
+      onClick: this.handleClick,
+      setButtonRef: this.setDropDownButtonRef,
+      expanded: this.state.buttonExpand,
+      icon: <Icon name="chevron" svgHeight={20} svgWidth={20} />
+    };
+    const dropdownClasses = classNames({
+      'ma__link-dropdown': true,
+      'ma__link-dropdown--expanded': this.state.buttonExpand
+    });
+    const dropdownMenuClasses = classNames({
+      'ma__link-dropdown-menu': true,
+      'ma__link-dropdown-menu--expanded': this.state.buttonExpand
+    });
+    const dropdownItemClasses = classNames({
+      'ma__link-dropdown-item': true,
+      'ma__link-dropdown-item--expanded': this.state.buttonExpand
+    });
+
+    return(
+      <div ref={this.wrapperRef} className={dropdownClasses}>
+        <ButtonWithIcon {...dropdownButtonProps} />
+        {this.state.buttonExpand && (
+          <div className={dropdownMenuClasses} aria-labelledby={dropdownButtonProps.id}>
+            {dropdownItems.map((item) => <DecorativeLink {...item} className={dropdownItemClasses} onClick={this.handleSelect} />)}
```

Comment:
this handleSelect function will never get triggered because onClick is not a prop for DecorativeLink. If we want to pass a custom click function here we should probably use Button